pavelib/assets.py

```diff
@@ -20,75 +20,174 @@
 
 ALL_SYSTEMS = ['lms', 'studio']
 COFFEE_DIRS = ['lms', 'cms', 'common']
-# A list of directories.  Each will be paired with a sibling /css directory.
-COMMON_SASS_DIRECTORIES = [
+
+LMS = 'lms'
+CMS = 'cms'
+
+SYSTEMS = {
+    'lms': LMS,
+    'cms': CMS,
+    'studio': CMS
+}
+
+# Common lookup paths that are added to the lookup paths for all sass compilations
+COMMON_LOOKUP_DIRS = [
+    path("common/static"),
     path("common/static/sass"),
 ]
-LMS_SASS_DIRECTORIES = [
-    path("lms/static/sass"),
-    path("lms/static/themed_sass"),
-    path("lms/static/certificates/sass"),
-]
-CMS_SASS_DIRECTORIES = [
-    path("cms/static/sass"),
-]
-THEME_SASS_DIRECTORIES = []
-SASS_LOAD_PATHS = ['common/static', 'common/static/sass']
-
-
-def configure_paths():
-    """Configure our paths based on settings.  Called immediately."""
-    edxapp_env = Env()
-    if edxapp_env.feature_flags.get('USE_CUSTOM_THEME', False):
-        theme_name = edxapp_env.env_tokens.get('THEME_NAME', '')
-        parent_dir = path(edxapp_env.REPO_ROOT).abspath().parent
-        theme_root = parent_dir / "themes" / theme_name
-        COFFEE_DIRS.append(theme_root)
-        sass_dir = theme_root / "static" / "sass"
-        css_dir = theme_root / "static" / "css"
-        if sass_dir.isdir():
-            css_dir.mkdir_p()
-            THEME_SASS_DIRECTORIES.append(sass_dir)
-
-    if edxapp_env.env_tokens.get("COMPREHENSIVE_THEME_DIR", ""):
-        theme_dir = path(edxapp_env.env_tokens["COMPREHENSIVE_THEME_DIR"])
-        lms_sass = theme_dir / "lms" / "static" / "sass"
-        lms_css = theme_dir / "lms" / "static" / "css"
-        if lms_sass.isdir():
-            lms_css.mkdir_p()
-            THEME_SASS_DIRECTORIES.append(lms_sass)
-        cms_sass = theme_dir / "cms" / "static" / "sass"
-        cms_css = theme_dir / "cms" / "static" / "css"
-        if cms_sass.isdir():
-            cms_css.mkdir_p()
-            THEME_SASS_DIRECTORIES.append(cms_sass)
-
-configure_paths()
-
-
-def applicable_sass_directories(systems=None):
-    """
-    Determine the applicable set of SASS directories to be
-    compiled for the specified list of systems.
-
-    Args:
-        systems: A list of systems (defaults to all)
-
-    Returns:
-        A list of SASS directories to be compiled.
-    """
-    if not systems:
-        systems = ALL_SYSTEMS
-    applicable_directories = []
-    applicable_directories.extend(COMMON_SASS_DIRECTORIES)
-    if "lms" in systems:
-        applicable_directories.extend(LMS_SASS_DIRECTORIES)
-    if "studio" in systems or "cms" in systems:
-        applicable_directories.extend(CMS_SASS_DIRECTORIES)
-    applicable_directories.extend(THEME_SASS_DIRECTORIES)
+
+# system specific lookup path additions, add sass dirs if one system depends on the sass files for other systems
+SASS_LOOKUP_DEPENDENCIES = {
+    'cms': [path('lms') / 'static' / 'sass' / 'partials', ],
+}
+
+THEME_SASS_WATCHER_DIRS = []
+
+
+def get_sass_directories(system, theme_dir):
```

Comment:
Nit: I'd make theme_dir optional defaulting to None, since all but one of the usages don't specify a theme.